jetty-core/jetty-proxy/src/test/java/org/eclipse/jetty/proxy/InterimResponseProxyTest.java

```diff
@@ -44,20 +44,10 @@ public void testInterimResponses() throws Exception
             public void process(Request request, Response response, Callback callback)
             {
                 CompletableFuture<Void> completable = response.writeInterim(HttpStatus.CONTINUE_100, HttpFields.EMPTY)
-                    .thenCompose(ignored ->
-                    {
-                        Promise.Completable<String> p = new Promise.Completable<>();
-                        Content.Source.asString(request, StandardCharsets.UTF_8, p);
-                        return p;
-                    })
+                    .thenCompose(ignored -> Promise.Completable.<String>with(p -> Content.Source.asString(request, StandardCharsets.UTF_8, p)))
```

Comment:
It really show here that a `Content.Source.asString` that returns a `CompletableFuture` would be more elegant.